### DISCUSSION TITLE
Callable generic type alias without parameter

### DISCUSSION THREAD

**Discussion by decorator-factory** [python/typing#1134] [Q&A] [answered]
# Question

When a generic callable type alias, like `MyFunction = Callable[[T], T]`, is used standalone, does it mean a generic function type of `Callable[[Any], Any]`? 

It seems to me that the latter should be used (to be consistent with generic classes, e.g. `list` means `list[Any]`). But Pyright disagrees. Is it an issue with pyright, or are both correct?

# Code
```py
from collections.abc import Callable
from typing import TypeVar

T = TypeVar("T")

def f1() -> Callable[[T], T]:
    ...
g1 = f1()
reveal_type(g1)
reveal_type(g1(42))

Hmm = Callable[[T], T]
def f2() -> Hmm:
    ...
g2 = f2()
reveal_type(g2)
reveal_type(g2(42))
```
# Pyright's (1.1.237) answer
```
information: Type of "g1" is "(T@f1) -> T@f1"
information: Type of "g1(42)" is "int"
information: Type of "g2" is "(T@Hmm) -> T@Hmm"
information: Type of "g2(42)" is "int"
```
# Mypy's (0.942) answer
```
main.py:9: note: Revealed type is "def [T] (T`-1) -> T`-1"
main.py:10: note: Revealed type is "builtins.int*"
main.py:16: note: Revealed type is "def (Any) -> Any"
main.py:17: note: Revealed type is "Any"
```

**Comment by erictraut** [comment, marked as answer]
I think this is an inconsistency (i.e. a bug) in pyright. It should mirror the behavior of mypy in this case. I'll fix it.

**Reply by erictraut** [reply]
I filed this tracking issue: https://github.com/microsoft/pyright/issues/3346
